mp4/lou.go

```diff
@@ -0,0 +1,279 @@
+	version := byte(versionAndFlags >> 24)
+	b := &loudnessBaseBox{
+		Version: version,
+		Flags:   versionAndFlags & flagsMask,
+	}
+	var loudnessBaseCount uint8
+	if b.Version >= 1 {
+		loudnessBaseCount = 0x3f & sr.ReadUint8()
+	} else {
+		loudnessBaseCount = 1
+	}
+
+	for a := uint8(0); a < loudnessBaseCount; a++ {
+		l := &LoudnessBase{}
+		if b.Version >= 1 {
+			l.EQSetID = 0x3f & sr.ReadUint8()
+		}
+		downmixIDAndDRCSetID := sr.ReadUint16()
+		l.DownmixID = uint8(downmixIDAndDRCSetID >> 6)
+		l.DRCSetID = uint8(downmixIDAndDRCSetID & 0x3f)
+		peakLevels := sr.ReadUint24()
+		l.BsSamplePeakLevel = int16(peakLevels >> 12)
+		l.BsTruePeakLevel = int16(peakLevels & 0x0fff)
+		measurementSystemAndReliablityForTP := sr.ReadUint8()
+		l.MeasurementSystemForTP = measurementSystemAndReliablityForTP >> 4
+		l.ReliabilityForTP = measurementSystemAndReliablityForTP & 0x0f
+		measurementCount := sr.ReadUint8()
+		for i := uint8(0); i < measurementCount; i++ {
+			m := &Measurement{}
+			m.MethodDefinition = sr.ReadUint8()
+			m.MethodValue = sr.ReadUint8()
+			measurementSystemAndReliablity := sr.ReadUint8()
+			m.MeasurementSystem = measurementSystemAndReliablity >> 4
+			m.Reliability = measurementSystemAndReliablity & 0x0f
+			l.Measurements = append(l.Measurements, m)
+		}
+		b.LoudnessBases = append(b.LoudnessBases, l)
+	}
+	return b, nil
+}
+
+func (b *loudnessBaseBox) size() uint64 {
+	size := 4
+	if b.Version >= 1 {
+		size += 1
+	}
+	for _, l := range b.LoudnessBases {
+		if b.Version >= 1 {
+			size += 8 + int(len(l.Measurements)*3)
+		} else {
+			size += 7 + int(len(l.Measurements)*3)
+		}
+	}
+	return uint64(size)
+}
+
+func (b *loudnessBaseBox) encode(w io.Writer) error {
+	sw := bits.NewFixedSliceWriter(int(b.size()))
+	err := b.encodeSW(sw)
+	if err != nil {
+		return err
+	}
+	_, err = w.Write(sw.Bytes())
+	return err
+}
+
+func (b *loudnessBaseBox) encodeSW(sw bits.SliceWriter) error {
+	versionAndFlags := (uint32(b.Version) << 24) + b.Flags
+	sw.WriteUint32(versionAndFlags)
+	if b.Version >= 1 {
+		sw.WriteUint8(0x3f & uint8(len(b.LoudnessBases)))
+	}
+	for a := 0; a < len(b.LoudnessBases); a++ {
+		l := b.LoudnessBases[a]
+		if b.Version >= 1 {
+			sw.WriteUint8(0x3f & l.EQSetID)
+		}
+		downmixIDAndDRCSetID := (uint16(l.DownmixID) << 6) | uint16(0x3f&l.DRCSetID)
+		sw.WriteUint16(downmixIDAndDRCSetID)
+		peakLevels := (uint32(l.BsSamplePeakLevel) << 12) | uint32(0x0fff&l.BsTruePeakLevel)
+		sw.WriteUint24(peakLevels)
+		measurementSystemAndReliablityForTP := (l.MeasurementSystemForTP << 4) | (0x0f & l.ReliabilityForTP)
+		sw.WriteUint8(measurementSystemAndReliablityForTP)
+		sw.WriteUint8(uint8(len(l.Measurements)))
+		for i := 0; i < len(l.Measurements); i++ {
+			m := l.Measurements[i]
+			sw.WriteUint8(m.MethodDefinition)
+			sw.WriteUint8(m.MethodValue)
+			measurementSystemAndReliablity := (m.MeasurementSystem << 4) | (0x0f & m.Reliability)
+			sw.WriteUint8(measurementSystemAndReliablity)
+		}
+	}
+	return nil
```

Comment:
Here you should return `sw.AccError()` since there might have occurred an error durin on of the `sw.Write...` calls.